### ISSUE TITLE
muiltiple copies of note open when using tasks dialogue

### ISSUE THREAD

**Issue by pcause** [Leonezz/obsidian-tasks-calendar-wrapper#18]
I have several todos in a single note.  When I use the edit button to edit the task, it also open the note.  When I do that for the next item it opens the note again.  I think this should reuse the note tab if open and not open multiple times.  Better yet, an option to not open the note at all would be good, but perhaps you have to do that to edit.

**Comment by Leonezz** [comment]
Do you mean your obsidian is opening multiple tabs for one same note file ? I don't know the reason if so cause it never happened in my environment. The behavior is as what you described, after the first item clicked and the note opened, the second click will reuse the opened tab.

You are right about the open to edit, the edit functionality is purely based on the Tasks plugin, I need to open the note and move cursor to the item first to let the Tasks plugin know where I want to edit.

**Comment by pcause** [comment]
Yes.  I have the note file already opened in a tab.  When I click the pencil a second tab opens (using 1.3.0 but happened before).  And each time I click the pencil I get another tab with the same note.

When I use a tasks query in my daily note, tasks doesn't open a new tab for the note containing the item.  DO you open the note or is it the tasks dialogue.  Perhaps you might file an issue on tasks about this so that there is a way to edit without opening or perhaps close the tab after editing is done.

**Comment by Leonezz** [comment]
The tab problem is weird, I will do some digging about it.

The answer is I am the one opening the note, it's the only way I know to pass information about the item to be edited to the Tasks api. The fact that the Tasks query edit icon does not require opening note is true and it is because the edit icon and the click handler of the icon is from inside the Tasks plugin, and apparently they have channels to communicate from the inside in an unconscious way, instead of through currently opened file and current cursor position as how I do it.

It will be much better if the Tasks api allows passing filename and item positions as arguments, I think this could a feature request for Tasks.

**Comment by Mayflower-breeze** [comment]
I read on the tasks website that it wouldn't recognize double spaced items. This plugin adds a blank line between each new entry. I was able to use the pencil to edit items, using the Task ui, by removing the double space on the note page.

**Comment by Leonezz** [comment]
> I read on the tasks website that it wouldn't recognize double spaced items. This plugin adds a blank line between each new entry. I was able to use the pencil to edit items, using the Task ui, by removing the double space on the note page.

Thanks for the information, I will check it out and make some modifications if it could be fixed.
